### DISCUSSION TITLE
Glitchy lines appearing inside containers.

### DISCUSSION THREAD

**Discussion by anthony-S93** [i3/i3#5212] [Q&A] [answered]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

![vlcsnap-2022-10-16-02h15m35s378](https://user-images.githubusercontent.com/69449791/196018329-14f669ce-6a58-471f-a9c6-8b3dfec61714.png)

![vlcsnap-2022-10-16-02h15m05s132](https://user-images.githubusercontent.com/69449791/196018337-1dd9cf03-8441-4b70-ab9b-dc73ece49a17.png)

![vlcsnap-2022-10-16-02h05m51s367](https://user-images.githubusercontent.com/69449791/196018343-348542b2-dd98-4fe5-b711-7b08962b456f.png)

This occurred (seemingly at random) when I: a) launch new windows b) switch between workspaces. So far, this occurred in Alacritty and some gtk apps (like thunar). Please note that the glitches appears only for a split second. Also note that disabling compositor did not solve the issue.

**UPDATE:**
After further testing, I noticed that these glitchy lines and patches are somehow linked to the graphical contents of the last-focused container. For example, I purposely switched to container whose background color was red. After that, when I launched an instance of Alacritty in another workspace, red lines and patches flashed in the window before it disappeared. 

## Expected Behavior

No glitches. 

## Reproduction Instructions

Switch between workspaces repeatedly until you see the glitch.  

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: 4.21 (2022-09-21)
</pre>

Graphics
<pre>
Graphics:
  Device-1: Intel 4th Gen Core Processor Integrated Graphics vendor: Lenovo
    driver: i915 v: kernel arch: Gen-5.75 ports: active: HDMI-A-1,VGA-1,eDP-1
    empty: DP-1,DP-2,HDMI-A-2 bus-ID: 00:02.0 chip-ID: 8086:0416
  Device-2: NVIDIA GK106GLM [Quadro K2100M] vendor: Lenovo driver: nouveau
    v: kernel arch: Kepler pcie: speed: 2.5 GT/s lanes: 16 ports: active: none
    empty: DVI-D-1,eDP-2 bus-ID: 01:00.0 chip-ID: 10de:11fc temp: 51.0 C
  Device-3: Chicony Integrated Camera type: USB driver: uvcvideo
    bus-ID: 3-12:3 chip-ID: 04f2:b39a
  Display: x11 server: X.Org v: 21.1.4 compositor: Picom v: git-7e568
    driver: X: loaded: intel,modesetting alternate: fbdev,nouveau,nv,vesa
    dri: i965,nouveau gpu: i915 display-ID: :0 screens: 1
  Screen-1: 0 s-res: 3840x2160 s-dpi: 96
  Monitor-1: HDMI-A-1 mapped: HDMI1 pos: top-right model: Acer KA222Q
    res: 1920x1080 dpi: 102 diag: 546mm (21.5")
  Monitor-2: VGA-1 mapped: VGA1 pos: primary,top-left model: Acer KA222Q
    res: 1920x1080 dpi: 102 diag: 546mm (21.5")
  Monitor-3: eDP-1 mapped: eDP1 pos: primary,bottom-l model: ChiMei InnoLux
    0x15c3 res: 1920x1080 dpi: 143 diag: 394mm (15.5")
  OpenGL: renderer: Mesa Intel HD Graphics 4600 (HSW GT2) v: 4.6 Mesa
    22.2.1 direct render: Yes
</pre>

<details><summary>Config file</summary><pre>
</pre>
</details>

<pre>
Logfile URL: 
</pre>

<pre>
- Linux Distribution & Version:  EndeavourOS Linux x86_64
- Are you using a compositor (e.g., xcompmgr or compton): picom (although the issue persists even after disabling the compositor)
</pre>

**Comment by stapelberg** [comment, marked as answer]
i3 isn’t involved in rendering window contents at all, it only arranges where windows should end up on the screen.

Hence, it’s very unlikely that i3 is responsible for these artifacts. Instead, more likely candidates are your graphic card driver, kernel, Xorg, or compositor.

You could also test with other window managers to further narrow down the problem (e.g. awesome).

**Reply by anthony-S93** [reply]
Indeed. After spending an entire day experimenting and testing, I managed to figure out what's causing the issue: the `xf86-video-intel` driver. 

Uninstalling that package fixed the issue.
